### PR TITLE
fix: multibase in pubsub http rpc  interop tests

### PR DESCRIPTION
**Pull request by lidel**
This PR updates `pubsub`  interop tests with  HTTP RPC wire format changes introduced in https://github.com/ipfs/js-ipfs/pull/3922 and   https://github.com/ipfs/go-ipfs/pull/8183 and runs against the code from mentioned PRs (js-ipfs from https://github.com/ipfs/js-ipfs/pull/3922 + go-ipfs 0.11.0-rc1) to ensure things work end-to-end.

## TODO
- [x] use js-ipfs and js-ipfs-http-client from https://github.com/ipfs/js-ipfs/pull/3922
- [x] use go-ipfs from https://github.com/ipfs/go-ipfs/pull/8183
- [x] make CI green with existing tests
- [x] update pubsub tests to use topic and data payload with `\n` to safeguard against regressions described in https://github.com/ipfs/go-ipfs/issues/7939
- [x] remove `http-rpc-runtime` or refactor it to be useful outside of this PR. The Rube Goldberg machine for testing HTTP RPC wire format change end-to-end
    it should be removed after:
  - [ ] js-ipfs release ships with necessary changes
  - [x] go-ipfs ships with necessary changes
- [x] rebase and restore tests skipped in https://github.com/ipfs/interop/pull/388

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -37,4 +37,5 @@ dist
 test/test-data/go-ipfs-repo/LOCK
 test/test-data/go-ipfs-repo/LOG
 test/test-data/go-ipfs-repo/LOG.old
-types
\ No newline at end of file
+types
+tmp
```
Comment (achingbrain) [inline .gitignore:41]: It looks like all the work happens in `/tmp` so is this ignore necessary?

File: test/pubsub.js [hunk 2]
```diff
@@ -100,8 +99,8 @@ describe.skip('pubsub', function () {
       })
 
       it('should exchange non ascii data', function () {
```
Comment (achingbrain) [inline test/pubsub.js:101]: Maybe the changes here should be a separate test?

File: test/pubsub.js [hunk 3]
```diff
@@ -127,7 +126,7 @@ describe.skip('pubsub', function () {
 
       it('should exchange binary data', function () {
         const data = uint8ArrayFromString('a36161636179656162830103056164a16466666666f400010203040506070809', 'base16')
-        const topic = 'pubsub-binary'
+        const topic = 'pubsub\nbinary\n'
```
Comment (achingbrain) [inline test/pubsub.js:129]: Same, this should be a separate test - we are testing binary data here, not newlines in topic names.

**Review by achingbrain — CHANGES_REQUESTED** [review]
LGTM, just a few small things in the comments that need addressing.

**Comment by lidel** [comment]
Rebased on top of #397

File: scripts/custom-runtime.sh [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+WORKDIR=$(pwd)
+
+if [ "$IPFS_GO_EXEC" == /tmp/go-ipfs/cmd/ipfs/ipfs ]; then
+if [ ! -d /tmp/go-ipfs ]; then
+    cd /tmp
+        git clone https://github.com/coryschwartz/go-ipfs.git
+        cd go-ipfs
+            # implementation from https://github.com/ipfs/go-ipfs/pull/8183
+            git checkout 153697d524f449ee9bec97245b0fcd7ebc2e8170
```
Comment (lidel) [inline scripts/custom-runtime.sh:28]: Switching to master in prep for https://github.com/ipfs/go-ipfs/pull/8183 to land.
```suggestion
        git clone https://github.com/ipfs/go-ipfs.git
        cd go-ipfs
            # implementation from https://github.com/ipfs/go-ipfs/pull/8183
            git switch master
```

**Comment by lidel** [comment]
Continued in https://github.com/ipfs/interop/pull/410

**Pull request closed without merging**